crates/volta-core/src/tool/package/mod.rs

```diff
@@ -182,20 +187,17 @@ fn remove_config_and_shim(bin_name: &str, pkg_name: &str) -> Fallible<()> {
 /// all the binaries installed by the input package.
 fn binaries_from_package(package: &str) -> Fallible<Vec<String>> {
     let bin_config_dir = volta_home()?.default_bin_dir();
-    if bin_config_dir.exists() {
-        dir_entry_match(&bin_config_dir, |entry| {
-            let path = entry.path();
-            if let Ok(config) = BinConfig::from_file(path) {
-                if config.package == package {
-                    return Some(config.name);
-                }
-            };
-            None
-        })
-        .with_context(|| ErrorKind::ReadBinConfigDirError {
-            dir: bin_config_dir.to_owned(),
-        })
-    } else {
-        Ok(vec![])
-    }
+    dir_entry_match(&bin_config_dir, |entry| {
+        let path = entry.path();
+        if let Ok(Some(config)) = BinConfig::from_file_if_exists(path) {
```

Comment:
Very minor, but since we're only matching on `Ok(Some(_))` here, I think we can leave this line as it was (since it amounts to the same thing):

```suggestion
        if let Ok(config) = BinConfig::from_file(path) {
```